backend/routes.go

```diff
@@ -73,6 +75,41 @@ func (s *server) handleAddUser() http.HandlerFunc {
 	}
 }
 
+//GET: endpoint for all articles
+func (s *server) handleGetArticles() http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		fmt.Fprintf(w, "Requesting list of all articles")
+
+		type ArticleJson struct {
+			Title string `json:"title"`
+			URL   string `json:"url"`
+			//Votes []Vote `json:"votes"`
+		}
+
+		count, results, err := s.db.GetAllArticles()
+
+		if err != nil {
+			s.respond(w, req, makeErrorResponse(err), 500)
+			return
+		}
+
+		var articles = make([]ArticleJson, count)
+		var article ArticleJson
+		for i, a := range results {
```

Comment:
If you apply the `append` way I'm suggesting, you would probably want to make this `i` ignorable, i.e:

```suggestion
		for _, a := range results {
```